src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -28,15 +29,10 @@
 
 MTROTAProviderDelegateBridge::~MTROTAProviderDelegateBridge(void) {}
 
-void MTROTAProviderDelegateBridge::setDelegate(id<MTROTAProviderDelegate> delegate, dispatch_queue_t queue)
+void MTROTAProviderDelegateBridge::setDelegate(id<MTROTAProviderDelegate> delegate)
 {
-    if (delegate && queue) {
-        mDelegate = delegate;
-        mQueue = queue;
-    } else {
-        mDelegate = nil;
-        mQueue = nil;
-    }
+    mDelegate = delegate ?: nil;
+    mQueue = chip::DeviceLayer::PlatformMgrImpl().GetWorkQueue();
```

Comment:
This means the delegate will get called back on the Matter queue, right?

At the very least this needs to be clearly documented in the API documentation (by saying that the delegate will get called back on some random queue and might need to proxy things over to the right queue itself).

Reply:
We are back to allowing the caller set the queue.